core/services/major_tom/major_tom.py

```diff
@@ -0,0 +1,96 @@
+#! /usr/bin/env python3
+import datetime
+import time
+import uuid
+from dataclasses import asdict, dataclass
+from typing import Any, Dict, Optional
+from zoneinfo import ZoneInfo
+
+from src.core import DefaultPayload, TelemetryEngine, get_latency
+from src.metrics import Metrics
+from src.typedefs import ExtensionInfo, VersionInfo
+
+LOG_SESSION_UUID = str(uuid.uuid4())
+
+SERVICE_NAME = "major_tom"
+LOG_FOLDER_PATH = f"/var/logs/blueos/services/{SERVICE_NAME}/"
+
+TELEMETRY_ENDPOINT = "https://telemetry.blueos.cloud/api/v1/anonymous/"
+S3_TELEMETRY_ENDPOINT = "https://telemetry.blueos.cloud/api/v1/anonymous/s3/"
+
+
+def get_hardware_uuid() -> str:
+    try:
+        with open("/etc/blueos/hardware-uuid", "r", encoding="utf-8") as f:
+            return f.read().strip()
+    except Exception as error:
+        print(f"unable to read machine id: {error}")
+        return ""
+
+
+def get_blueos_uuid() -> str:
+    try:
+        with open("/etc/blueos/uuid", "r", encoding="utf-8") as f:
+            return f.read().strip()
+    except Exception as error:
+        print(f"unable to read blueos id: {error}")
```

Comment:
We should add loguru logging.

Reply:
I'll look into this, this might be tricky as we are using loguru for managing the telemetry